salt/states/file.py

```diff
@@ -406,16 +539,22 @@ def _process(name):
                 # another state.
                 if low['name'] == comp['file'] or low['__id__'] == comp['file']:
                     fn = low['name']
+                    fun = low['fun']
                     if os.path.isdir(fn):
                         if _is_child(fn, name):
-                            if walk_d:
+                            if fun == 'recurse':
+                                fkeep = _gen_recurse_managed_files(**low)[3]
+                                log.debug('Keep from {0}: {1}'.format(fn, fkeep))
+                                keep.update(fkeep)
+                            elif walk_d:
                                 walk_ret = set()
                                 _process_by_walk_d(fn, walk_ret)
                                 keep.update(walk_ret)
                             else:
                                 keep.update(_process(fn))
                     else:
                         keep.add(fn)
+    log.debug('Files to keep from required states: %s' % list(keep))
```

Comment:
The logging functions already perform string replacement using the single-asterisk expansion. Moreover, I think our pylint currently flags usage of the ``%`` operator. This line should be restated as:

```python
log.debug('Files to keep from required states: %s', list(keep))
```